plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6802,6 +6813,414 @@ public void testUpdatingFileFormat()
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'ORC', compression_level = 5", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'PARQUET'", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'ORC'", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES write_compression = 'SNAPPY'", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'PARQUET', write_compression = 'SNAPPY'", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'PARQUET', compression_level = 5", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'ORC', write_compression = 'SNAPPY'", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'ORC', compression_level = 5", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertQueryFails(format("ALTER TABLE %s SET PROPERTIES format = 'PARQUET', write_compression = 'SNAPPY', compression_level = 5", tableName),
+                "File format .* with compression codec .* and compression level .* not supported");
+
+        assertUpdate(format("DROP TABLE %s", tableName));
+
+    }
+
+    @Test
+    public void testUpdatingFileFormatCompressionMixed()
+    {
+        String tableName = "test_updating_file_format_compression_mixed" + randomNameSuffix();
+
+        assertUpdate(format("CREATE TABLE %s WITH (format = 'AVRO') AS SELECT * FROM nation", tableName), "SELECT count(*) FROM nation");
+
+        assertUpdate(format("ALTER TABLE %s SET PROPERTIES write_compression = 'ZSTD'", tableName));
+        assertQuery(format("SELECT value FROM \"%s$properties\" WHERE key = 'write.format.default'", tableName), format("VALUES '%s'", AVRO));
+        assertQuery(format("SELECT value FROM \"%s$properties\" WHERE key = 'write.avro.compression-codec'", tableName), "VALUES 'ZSTD'");
+
+        assertUpdate(format("ALTER TABLE %s SET PROPERTIES compression_level = 5", tableName));
+        assertQuery(format("SELECT value FROM \"%s$properties\" WHERE key = 'write.avro.compression-level'", tableName), ("VALUES 5"));
```

Comment:
Looks like checkstyle is complaining about this line, but giving the wrong line number in the error.